File: app/billing/policy.py
from __future__ import annotations

PAID_PLANS = {"pro", "admin"}
PRO_EXPORT_FORMATS = {"pdf", "docx"}


def can_export_format(user_plan: str | None, export_format: str) -> bool:
    plan = (user_plan or "free").strip().lower()
    fmt = (export_format or "").strip().lower()
    if fmt not in PRO_EXPORT_FORMATS:
        return True
    return plan in PAID_PLANS

File: frontend/src/features/entitlements/policy.ts
export type Plan = 'free' | 'pro' | 'admin' | null | undefined;
export type ExportFormat = 'txt' | 'srt' | 'vtt' | 'json' | 'pdf';

export function canExportFormat(_args: { plan: Plan; format: ExportFormat }): boolean {
  return true;
}
